### ISSUE TITLE
"Attempting to call unbound fn" with require-sql

### ISSUE THREAD

**Issue by kendagriff** [krisajenkins/yesql#118]
In moving to 0.5.1, `require-sql` seemed to be the best way to support the reloaded pattern. Not sure if I'm doing something wrong, but I'm at a loss for how to proceed after modifying our code to support the new function signatures:

```
Testing id.login-test
java.lang.IllegalStateException: Attempting to call unbound fn: #'yesquire/sql/legacy.sql/get-login-details
at clojure.lang.Var$Unbound.throwArity (Var.java:43)
clojure.lang.AFn.invoke (AFn.java:36)
id.store.legacy.Legacy.get_login_details (legacy.clj:19)
...
```

Here's the SQL:

```
--name: get-login-details
select id, username...
from users
where username = :username
limit 1
```

Here's the relevant Clojure code:
```clojure
(require-sql ["sql/legacy.sql" :as legacy])
...
(legacy/get-login-details {:username username} {:connection (:spec db)})
```

Also, the function documentation seems to conflict with the README:

```clojure
yesquire/sql/legacy.sql/get-login-details
([{:keys (username)} {:keys [connection]}])
```

Docs:
```clojure
(user/users-by-country db-spec "GB")
;=> ({:count 58})
```

Note the first argument is the connection, the second a `?` style argument.

Thoughts?

**Comment by xeqi** [comment]
I also ran into this. Ended up making an empty namespace and using `defqueries` as a work around to keep the functions separate.

```clojure
(ns proj.db.sql
  (:require [yesql.core :refer [defqueries]]))

(defqueries "queries/sqlite-queryfile.sql")
```

**Comment by timgluz** [comment]
i encountered similar issue when tried to use `require-sql` on REPL.

Maybe @iantruslove as commit's author knows more about it.

**Comment by kendagriff** [comment]
If I understand correctly, `require-sql` is the only way to pass a DB connection to the query functions *at runtime* in the newer version of yesql, right?

**Comment by flsafe** [comment]
@kendagriff  I think I've run into the same issue. `require-sql` is not the only way to provide a db connection to the query function at run time. I believe if you look at the code, you can provide one on query function invocation: https://github.com/krisajenkins/yesql/blob/master/src/yesql/generate.clj#L113

**Comment by kendagriff** [comment]
@flsafe Yep, you're right! It's documented in the [Migration Guide](https://github.com/krisajenkins/yesql/wiki/Migration#step-1).

**Comment by stig** [comment]
I ran into this issue as well.

**Comment by atstp** [comment]
Judging by the thread so far, it looks like as of `0.5.1`, `require-sql` doesn't work.  Is this correct?

**Comment by iantruslove** [comment]
I don't think require-sql works with the rest of the code in 0.5.1

**Comment by atstp** [comment]
Thanks!

**Comment by sandaemc** [comment]
Still broken. I think the solution is to just add the :connection map to `require-sql`?
